File: 2ndSemester/AlgorithimsAndDataStructuresII/Practice/Ex intro/06 - Palíndromo em C-C++/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <stdbool.h>

bool checkPalindromo(char* palavra){
  bool resp = true;
  int tamanho = strlen(palavra);
  int i = 0;

  do{
    if(palavra[i] != palavra[tamanho - i -1]){
      resp = false;
    }
    i++;
  }
  while(i < (tamanho/2) && resp!= false);
  return resp;
}

int main(){
  char entrada[100];
  do{
    scanf(" %[^\n]s",entrada);
    if(checkPalindromo(entrada) == true){
      printf("SIM\n");
    }
    else{
      printf("NAO\n");
    }
  }while(strcmp(entrada, "FIM")!=0);
  return 0;
}

File: 2ndSemester/AlgorithimsAndDataStructuresII/U01-WarmingUp/05 - Palíndromo em Java/Palindromo.java
import java.util.Scanner;

class Palindromo{
	public static Scanner sc = new Scanner(System.in);	

	//metodo que determina a existencia de palindromo
	public static boolean checkPalindromo(String palavra){
		boolean resultado = true;
		int tamanho = palavra.length();
		int i = 0;

		do{
			if(palavra.charAt(i) != palavra.charAt(tamanho - 1 - i))
				resultado = false;
			i++;
		}while(i < tamanho/2 && resultado != false);
		return resultado;
	}

	//metodo main
 	public static void main(String[] args){
		String entrada;
		do{
			entrada = sc.nextLine();
			if(checkPalindromo(entrada)){
				System.out.println("SIM");
			}
			else{
				System.out.println("NAO");
			}
		}while(entrada.compareTo("FIM") != 0);
	}
}